File: library/src/main/java/me/shkschneider/skeleton/helper/BundleHelper.java
package me.shkschneider.skeleton.helper;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.io.Serializable;

public class BundleHelper {

    protected BundleHelper() {
        // Empty
    }

    public static Bundle pack(@NonNull final String key, @Nullable final Serializable serializable) {
        final Bundle bundle = new Bundle();
        if (serializable != null) {
            bundle.putSerializable(key, serializable);
        }
        return bundle;
    }

    @Nullable
    public static Serializable unpack(@NonNull final Bundle bundle, @NonNull final String key) {
        if (! bundle.containsKey(key)) {
            LogHelper.w("Bundle has no such key");
            return null;
        }

        return bundle.getSerializable(key);
    }

}

File: library/src/main/java/me/shkschneider/skeleton/helper/BundleHelper.kt
package me.shkschneider.skeleton.helper

import android.os.Bundle
import me.shkschneider.skeleton.helperx.Logger

import java.io.Serializable

object BundleHelper {

    fun pack(bundle: Bundle, key: String, serializable: Serializable?): Bundle {
        bundle.putSerializable(key, serializable)
        return bundle
    }

    fun pack(key: String, serializable: Serializable?): Bundle {
        val bundle = Bundle()
        return pack(bundle, key, serializable)
    }

    fun unpack(bundle: Bundle, key: String): Serializable? {
        bundle.getSerializable(key)?.let {
            return true
        }
        Logger.warning("Bundle has no such key")
        return null
    }

}
